src/main/java/org/jboss/remoting3/remote/RemoteReadListener.java

```diff
@@ -50,7 +50,7 @@
  */
 final class RemoteReadListener implements ChannelListener<ConduitStreamSourceChannel> {
     // FIXME temporarily using a system property as a solution to ACK TIMEOUT issue until an RFE is properly submitted
-    private static final int MESSAGE_ACK_TIMEOUT = Integer.parseInt(WildFlySecurityManager.getPropertyPrivileged("org.jboss.remoting3.remote.message.ack.timeout", String.valueOf(RemotingOptions.DEFAULT_MESSAGE_ACK_TIMEOUT)));
+    private static final long MESSAGE_ACK_TIMEOUT = Integer.parseInt(WildFlySecurityManager.getPropertyPrivileged("org.jboss.remoting3.remote.message.ack.timeout", String.valueOf(RemotingOptions.DEFAULT_MESSAGE_ACK_TIMEOUT))) * 1_000_000;
```

Comment:
This is subject to a wrapping error if the timeout is more than around 2000 milliseconds; maybe do it like this instead:

```suggestion
    private static final long MESSAGE_ACK_TIMEOUT = Integer.parseInt(WildFlySecurityManager.getPropertyPrivileged("org.jboss.remoting3.remote.message.ack.timeout", String.valueOf(RemotingOptions.DEFAULT_MESSAGE_ACK_TIMEOUT))) * 1_000_000L;
```